module/VuFind/src/VuFind/ILS/Driver/Voyager.php

```diff
@@ -187,7 +186,29 @@ protected function getDb()
                      ')' .
                    ')';
             try {
-                $this->lazyDb = new Oci8(
+                if ((!defined('PHP_MAJOR_VERSION ') || PHP_MAJOR_VERSION >= 8)
+                    && empty($this->config['Catalog']['forceOCI8Support'])
+                ) {
+                    $this->error(
+                        <<<EOT
+Voyager connection is only supported on PHP 7 by default. To enable support, you
+will need to manually update the yajra/laravel-pdo-via-oci8 package using the
+following command:
+
+php [path/to/]composer.phar update yajra/laravel-pdo-via-oci8 --ignore-platform-reqs
+
+Then force the Voyager driver to connect by adding the following setting to
+Voyager.ini or VoyagerRestful.ini:
+
+[Catalog]
+forceOCI8Support = true
+
+EOT
+                    );
+                    throw new ILSException('Unsupported PHP version');
+                }
+                // @phpstan-ignore-next-line
```

Comment:
Since you're ignoring the whole file via PHPstan configuration, is it worth removing this line and reverting the removal of the use statement, to minimize diffs? (I'm happy to make the changes while testing this, if that would save you time).

Reply:
The line can be removed, but I'm not sure about the use statement. Will PHP 8 throw a fatal when including the file which wouldn't give us the chance to write the log message?

Reply:
Use statements have no run-time impact; they just set up aliases that are used by the language parser. That's why they always have to come before executable code. A use statement on its own won't load anything, so I think the code should be functionally equivalent either way.

Reply:
In that case, sure. I'd appreciate it if you could make the changes. :)

Reply:
Will do!